packages/api/src/api/WidgetApiImpl.ts

```diff
@@ -441,12 +451,15 @@ export class WidgetApiImpl implements WidgetApi {
       roomIds,
     }: { messageType?: string; roomIds?: string[] | Symbols.AnyRoom } = {},
   ): Promise<Array<RoomEvent<T>>> {
-    return (await this.matrixWidgetApi.readRoomEvents(
+    const unvalidatedEvents = (await this.matrixWidgetApi.readRoomEvents(
       eventType,
       Number.MAX_SAFE_INTEGER,
       messageType,
       typeof roomIds === 'string' ? [Symbols.AnyRoom] : roomIds,
     )) as RoomEvent<T>[];
+
+    const validatedEvents = unvalidatedEvents.filter(isValidRoomEvent);
```

Comment:
Note for another day which I wouldn't block this on: Should we somehow warn or error at least in the console if an event is considered invalid? Basically, for debugging purposes when something breaks down the line.